### PR TITLE
DEVX-2151: Validate and remove ssl.endpoint.identification.algorithm …

### PR DESCRIPTION
**Pull request by ybyzek**
…from examples

Description:

According to the conversation in confluentinc/examples#820

> Remove ssl.endpoint.identification.algorithm as it matches the default since Apache Kafka 2.0 and it's not strictly required (i.e. things work without it).

I will validate all examples

- [x] https://github.com/confluentinc/examples/tree/6.0.0-post/ccloud/beginner-cloud
- [x] https://github.com/confluentinc/examples/tree/6.0.0-post/ccloud
- [x] https://github.com/confluentinc/examples/tree/6.0.0-post/clients/cloud/ksql-datagen

cc: @confluentinc/integration-architecture for FYI

### CODE REVIEW

File: clients/cloud/kafka-connect-datagen/start-docker-avro.sh [hunk 1]
```diff
@@ -22,4 +22,3 @@ sleep 90
...
 source ./submit_datagen_orders_config_avro.sh
 #docker-compose exec connect bash -c 'kafka-avro-console-consumer --topic test2 --bootstrap-server $CONNECT_BOOTSTRAP_SERVERS --consumer.config /tmp/ak-tools-ccloud.delta --property basic.auth.credentials.source=$CONNECT_VALUE_CONVERTER_BASIC_AUTH_CREDENTIALS_SOURCE --property schema.registry.basic.auth.user.info=$CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO --property schema.registry.url=$CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_URL --max-messages 5'
-docker-compose exec connect bash -c 'kafka-avro-console-consumer --topic test2 --bootstrap-server $CONNECT_BOOTSTRAP_SERVERS --consumer-property ssl.endpoint.identification.algorithm=https --consumer-property sasl.mechanism=PLAIN --consumer-property security.protocol=SASL_SSL --consumer-property sasl.jaas.config="$SASL_JAAS_CONFIG_PROPERTY_FORMAT" --property basic.auth.credentials.source=$CONNECT_VALUE_CONVERTER_BASIC_AUTH_CREDENTIALS_SOURCE --property schema.registry.basic.auth.user.info=$CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO --property schema.registry.url=$CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_URL --max-messages 5'
```
Comment (rspurgeon) [inline clients/cloud/kafka-connect-datagen/start-docker-avro.sh]: @ybyzek it's difficult to see how this removal is related to DEVX-2151

File: clients/cloud/kafka-connect-datagen/start-docker.sh [hunk 2]
```diff
@@ -22,4 +22,3 @@ sleep 90
 docker-compose exec connect bash -c 'kafka-topics --bootstrap-server `grep "^\s*bootstrap.server" /tmp/ak-tools-ccloud.delta | tail -1` --command-config /tmp/ak-tools-ccloud.delta --topic test1 --create --replication-factor 3 --partitions 6'
 source ./submit_datagen_orders_config.sh
 #docker-compose exec connect bash -c 'kafka-console-consumer --topic test1 --bootstrap-server $CONNECT_BOOTSTRAP_SERVERS --consumer.config /tmp/ak-tools-ccloud.delta --max-messages 5'
-docker-compose exec connect bash -c 'kafka-console-consumer --topic test1 --bootstrap-server $CONNECT_BOOTSTRAP_SERVERS --consumer-property ssl.endpoint.identification.algorithm=https --consumer-property sasl.mechanism=PLAIN --consumer-property security.protocol=SASL_SSL --consumer-property sasl.jaas.config="$SASL_JAAS_CONFIG_PROPERTY_FORMAT" --max-messages 5'
```
Comment (rspurgeon) [inline clients/cloud/kafka-connect-datagen/start-docker.sh]: @ybyzek it's difficult to see how this removal is related to DEVX-2151

File: clients/cloud/ksql-datagen/docker-compose.yml [hunk 3]
```diff
@@ -22,12 +22,10 @@ services:
     command: "bash -c '/usr/bin/ksql-datagen quickstart=orders format=json topic=test1 msgRate=2 bootstrap-server=$BOOTSTRAP_SERVERS propertiesFile=/tmp/ksql-datagen.delta'"
     environment:
       KSQL_CONFIG_DIR: "/etc/ksql"
-      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
```
Comment (rspurgeon) [inline clients/cloud/ksql-datagen/docker-compose.yml]: @ybyzek it's difficult to see how this removal is related to DEVX-2151

File: clients/cloud/ksql-datagen/docker-compose.yml [hunk 4]
```diff
@@ -37,12 +35,10 @@ services:
     command: "bash -c '/usr/bin/ksql-datagen quickstart=orders format=avro topic=test2 msgRate=2 schemaRegistryUrl=$SCHEMA_REGISTRY_URL bootstrap-server=$BOOTSTRAP_SERVERS propertiesFile=/tmp/ksql-datagen.delta'"
     environment:
       KSQL_CONFIG_DIR: "/etc/ksql"
-      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j-rolling.properties"
```
Comment (rspurgeon) [inline clients/cloud/ksql-datagen/docker-compose.yml]: @ybyzek it's difficult to see how this removal is related to DEVX-2151

File: clients/docs/ksql-datagen.rst [hunk 5]
```diff
@@ -99,7 +99,6 @@ Consume Records
 
       .. code-block:: bash
 
-         docker-compose exec connect bash -c 'kafka-console-consumer --topic test1 --bootstrap-server $CONNECT_BOOTSTRAP_SERVERS --consumer-property ssl.endpoint.identification.algorithm=https --consumer-property sasl.mechanism=PLAIN --consumer-property security.protocol=SASL_SSL --consumer-property sasl.jaas.config="$SASL_JAAS_CONFIG_PROPERTY_FORMAT" --max-messages 5'
```
Comment (rspurgeon) [inline clients/docs/ksql-datagen.rst]: @ybyzek it's difficult to see how this removal is related to DEVX-2151, but if it's being removed, should the code-block directive above it be removed as well?

File: clients/docs/ksql-datagen.rst [hunk 6]
```diff
@@ -172,7 +171,6 @@ Consume Avro Records
 
       .. code-block:: bash
 
-         docker-compose exec connect bash -c 'kafka-avro-console-consumer --topic test2 --bootstrap-server $CONNECT_BOOTSTRAP_SERVERS --consumer-property ssl.endpoint.identification.algorithm=https --consumer-property sasl.mechanism=PLAIN --consumer-property security.protocol=SASL_SSL --consumer-property sasl.jaas.config="$SASL_JAAS_CONFIG_PROPERTY_FORMAT" --property basic.auth.credentials.source=$CONNECT_VALUE_CONVERTER_BASIC_AUTH_CREDENTIALS_SOURCE --property schema.registry.basic.auth.user.info=$CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_BASIC_AUTH_USER_INFO --property schema.registry.url=$CONNECT_VALUE_CONVERTER_SCHEMA_REGISTRY_URL --max-messages 5'
```
Comment (rspurgeon) [inline clients/docs/ksql-datagen.rst]: @ybyzek it's difficult to see how this removal is related to DEVX-2151, but if it's being removed, should the code-block directive above it be removed as well?

Comment (ybyzek) [reply clients/docs/ksql-datagen.rst]: @rspurgeon in trying to validate this PR against the ksql-datagen example, I discovered another issue (`KSQL-3812: remove log4j.properties setting for ksql-datagen`) that I rolled into this PR.  But it makes for a confusing PR & review process, so I will revert the change and submit in a new PR.

File: clients/docs/ksql-datagen.rst (hunk 5 above)
Comment (ybyzek) [reply clients/docs/ksql-datagen.rst]: My script was too aggressive in deleting lines...fixing now

Comment (ybyzek) [reply clients/docs/ksql-datagen.rst]: And good catch ;)

**Review by rspurgeon — APPROVED** [review]
LGTM, @ybyzek I didn't verify the affected demos but did scan the code for only applicable line removals.

**Pull request merged**